pages/user/[username]/notifications.tsx

```diff
@@ -0,0 +1,230 @@
+      headers: {
+        accept: "application/json",
+        Authorization: `Bearer ${sessionToken}`,
+      },
+    });
+    const notifications = await req.json();
+    setNotificationsResponse(notifications);
+    setLoading(false);
+  };
+
+  const notifications = useMemo(() => {
+    if (!notificationsResponse?.data) return [];
+    if (filter === "all")
+      return notificationsResponse?.data?.sort(
+        (a, b) => new Date(b.notified_at).getTime() - new Date(a.notified_at).getTime()
+      );
+    return notificationsResponse?.data
+      .filter((notification) => notification.type === filter)
+      ?.sort((a, b) => new Date(b.notified_at).getTime() - new Date(a.notified_at).getTime());
+  }, [notificationsResponse?.data, filter]);
+
+  useEffect(() => {
+    fetchNotifications();
+    fetchUserData();
+
+    if (username && sessionToken) {
+      router.push(`/user/${username}/notifications?filter=all`);
+    }
+  }, [sessionToken, username]);
+
+  const handlePageChange = (page: number) => {
+    setPage(page);
+    fetchNotifications(page);
+    if (topRef.current) {
+      topRef.current.scrollIntoView({
+        behavior: "smooth",
+      });
+    }
+  };
+
+  return (
+    <div className="w-full ">
+      <TopNav />
+      <div
+        className="container flex flex-col justify-between w-full px-2 pt-16 mx-auto overflow-hidden md:px-16 md:flex-row md:gap-20 lg:gap-40"
+        ref={topRef}
+      >
+        <div className="flex flex-col gap-4 w-80 ">
+          <Title className="!text-2xl !text-light-slate-12" level={3}>
+            Notifications
+          </Title>
+          <aside className="flex flex-col gap-2">
+            <Button
+              variant="link"
+              className={clsx("hover:text-orange-600", filter === "all" ? "text-orange-600" : "text-light-slate-11")}
+              onClick={() => {
+                setFilter("all");
+                router.push(`/user/${username}/notifications?filter=all`);
+              }}
+            >
+              All
+            </Button>
+            <Button
+              variant="link"
+              className={clsx("hover:text-orange-600", filter === "follow" ? "text-orange-600" : "text-light-slate-11")}
+              onClick={() => {
+                setFilter("follow");
+                router.push(`/user/${username}/notifications?filter=follow`);
+              }}
+            >
+              Follow
+            </Button>
+            <Button
+              variant="link"
+              className={clsx(
+                "hover:text-orange-600",
+                filter === "highlight_reaction" ? "text-orange-600" : "text-light-slate-11"
+              )}
+              onClick={() => {
+                setFilter("highlight_reaction");
+                router.push(`/user/${username}/notifications?filter=highlight_reaction`);
+              }}
+            >
+              Highlight Reaction
```

Comment:
```suggestion
             Reaction
```
Reactions won't always be highlight specific